bothan-cryptocompare/src/worker/asset_worker.rs

```diff
@@ -0,0 +1,132 @@
+use std::sync::Weak;
+use std::time::Duration;
+
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::Decimal;
+use tokio::time::{interval, timeout};
+use tracing::{debug, error, info, warn};
+
+use bothan_core::store::WorkerStore;
+use bothan_core::types::AssetInfo;
+
+use crate::api::rest::CryptoCompareRestAPI;
+use crate::worker::error::ParseError;
+use crate::worker::CryptoCompareWorker;
+
+pub(crate) fn start_asset_worker(
+    weak_worker: Weak<CryptoCompareWorker>,
+    update_interval: Duration,
+) {
+    let mut interval = interval(update_interval);
+    tokio::spawn(async move {
+        while let Some(worker) = weak_worker.upgrade() {
+            info!("updating asset info");
+
+            let ids = match worker.store.get_query_ids().await {
+                Ok(ids) => ids.into_iter().collect::<Vec<String>>(),
+                Err(e) => {
+                    error!("failed to get query ids with error: {}", e);
+                    Vec::new()
+                }
+            };
+
+            let result = timeout(
+                interval.period(),
+                update_asset_info(&worker.store, &worker.api, &ids),
+            )
+            .await;
+
+            if result.is_err() {
+                warn!("updating interval exceeded timeout")
+            }
```

Comment:
pls make this more succinct if the result isn't going to be used

Reply:
how na?? result is used to be check if it is err and this is code from coingecko source

Reply:
```
            let update_fn = update_asset_info(&worker.store, &worker.api, &ids);
            if timeout(interval.period(), update_fn).await.is_err() {
                warn!("updating interval exceeded timeout")
            }
            ```